### PR TITLE
Update index.rst

### PR DESCRIPTION
**Pull request by petersmythe**
Add link to Swagger spec

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/user/source/rest/api/index.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
 REST configuration API reference
 ================================
 
-This section describes the GeoServer REST configuration API.
+This section describes the GeoServer REST configuration API.  You may also want to refer to the `Swagger spec <https://docs.geoserver.org/latest/en/user/rest/index.html#api>`_ but heed the warning.
```
Comment (jodygarnett) [inline doc/en/user/source/rest/api/index.rst:6]: What does this provided? reading ...

Better to repeat the what is there to set expectations ...

Comment (jodygarnett) [inline doc/en/user/source/rest/api/index.rst:6]:
```suggestion
This section describes the GeoServer REST configuration API. 

.. note:: You may also wish to review the hand-written :doc:`Open API  </rest/index>`_ documentation. This documentation not auto-generated and may not exactly match the latest GeoServer. Please treat this resource as documentation only as this is not suitable for client generation.

```

**Review by bradh — APPROVED** [review]

**Comment by bradh** [comment]
The change looks OK, but the docs build is failing with this:

```
/home/runner/work/geoserver/geoserver/doc/en/user/source/rest/api/index.rst:8: WARNING: Mismatch: both interpreted text role prefix and reference suffix.
```

**Comment by petersmythe** [comment]
Doc CI is now passing

**Comment by mprins** [comment]
for some reason the POM sorting check is tripping up; I will add a commit for that

**Comment by petersmythe** [comment]
> for some reason the POM sorting check is tripping up; I will add a commit for that

What is this supposed to do?  It looks like there was no change?

**Pull request merged**